nixos/modules/services/x11/display-managers/default.nix

```
@@ -77,8 +77,12 @@ let
       # Speed up application start by 50-150ms according to
       # http://kdemonkey.blogspot.nl/2008/04/magic-trick.html
       compose_cache="''${XCOMPOSECACHE:-$HOME/.compose-cache}"
-      rm -rf "$compose_cache"
       mkdir -p "$compose_cache"
+      # To avoid accidentally deleting a wrongly set up XCOMPOSECACHE directory,
+      # defensively try to delete cache *files* only, following the file format specified in
+      # https://gitlab.freedesktop.org/xorg/lib/libx11/-/blob/master/modules/im/ximcp/imLcIm.c#L330
+      # sprintf (*res, "%s/%c%d_%03x_%08x_%08x", dir, _XimGetMyEndian(), XIM_CACHE_VERSION, (unsigned int)sizeof (DefTree), hash, hash2);
+      find "$compose_cache" -maxdepth 1 -regex '.*/[Bl][0-9]+_[0-9]+_[0-9a-f]+_[0-9a-f]+' -delete
```

I think you have to use `find` from `findutils` to make sure that you have GNU find.  Also the third field is hex too.
```suggestion
      ${findutils}/bin/find "$compose_cache" -maxdepth 1 -regex '\./[Bl][0-9]+_[0-9a-f]+_[0-9a-f]+_[0-9a-f]+' -delete
```

Since we have to use GNU find anyway, we can make it even more concise using POSIX extended regular expression where we can match the exact number of characters, although that is probably overkill.
```suggestion
      ${findutils}/bin/find "$compose_cache" -maxdepth 1 -regextype posix-extended -regex '.*/[Bl][0-9]+_[0-9a-f]{3}_[0-9a-f]{8}_[0-9a-f]{8}' -delete
```